server/aesdsocket.c

```diff
@@ -0,0 +1,328 @@
+// File: aesdsocket.c
+// Description: Socket based program, receives data over connection and appends to file vat/tmp/aesdsocketdata. 
+// Author: Sachin Mathad
+// Date: 2/15/2023
+// ref: Lecture videos, Linux System programming and beej.us
+
+#include<stdint.h>
+#include<stdbool.h>
+#include<stdio.h>
+#include<syslog.h>
+#include<errno.h>
+#include<fcntl.h>
+#include<sys/stat.h>
+#include<sys/types.h>
+#include<unistd.h>
+#include<string.h>
+#include<stdlib.h>
+#include<signal.h>
+#include<arpa/inet.h>
+#include<sys/socket.h>
+#include<netinet/in.h>
+#include<netdb.h>
+#include<linux/fs.h>
+
+
+#define INIT_BUFFER_SIZE 1024
+
+// socket descriptor 
+int sfd; 
+// accept socket descriptor 
+int client_fd = -1;
+// file descriptor 
+int fd;
+
+char *store_buffer = NULL;
+
+// reentrant function
+void handler(int sig, siginfo_t *info, void *ucontext) {
+
+    if((sig == SIGINT) || (sig == SIGTERM)) {
+    
+        syslog(LOG_DEBUG, "Caught Signal, Exiting\n");
+        closelog();
+        close(client_fd);
+        close(sfd);
+        close(fd);
+        closelog();
+        unlink("/var/tmp/aesdsocketdata");
+        _exit(EXIT_SUCCESS);
```

Comment:
While this technically does meet the requirement to avoid reentrant functions I would not suggest this approach for your signal handler.  It will fall apart when cleanup becomes more complicated.  It also means you have two different cleanup paths depending on how your program exits.

Reply:
I have handled it with a flag in A6 release.